cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminOpHTTPHandler.java

```diff
@@ -193,4 +201,13 @@ public void upgrade(HttpRequest request, HttpResponder responder,
   private String getAdminOpErrorMessage(String opName, String instanceName) {
     return String.format("Error executing admin operation %s for dataset instance %s", opName, instanceName);
   }
+
+  // propagate userid from the HTTP Request in the current thread
+  private void propagateUserId(HttpRequest request) {
```

Comment:
haven't given it much thought, yet, but it seems to me that each such internal HTTP call should have a pipeline like the router, where one of the stages does this? We get this for free with externally originated calls that go via router, but nothing of that sort exists for inter-system-service communication.

Another thought is also that there needs to be a much more thorough way of CDAP system services authenticating with each other, but I have no details at all beyond that at this point.